girder/utility/s3_assetstore_adapter.py

```diff
@@ -73,7 +73,12 @@ def validateInfo(doc):
         client = S3AssetstoreAdapter._s3Client(params)
         if doc.get('readOnly'):
             try:
-                client.head_bucket(Bucket=doc['bucket'])
+                paginator = client.get_paginator('list_objects')
+                pageIterator = paginator.paginate(
+                    Bucket=doc['bucket'], Prefix=doc['prefix'], Delimiter='/')
+                for resp in pageIterator:
+                    resp.get('Contents', [])
```

Comment:
Why is this line here? Is this not sufficient:

```python
client.list_objects(Bucket=doc['bucket'], Prefix=doc['prefix'], Delimiter='/', MaxKeys=1)
```

?

Reply:
It probably is.  Sadly, I copied what we do when we enumerate items in a bucket/prefix, rather than figuring out the best way to do it.

Reply:
See https://github.com/girder/girder/pull/3440